From d2ddafbf815af449e135f6cd2a3958f7f91e5eea Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 4 Feb 2021 15:13:50 +0800
Subject: [PATCH] policy/modules/system/systemd: make systemd_backlight_t
 domain MLS trusted for read/write to files up to its clearance

Fixes:
avc:  denied  { search } for  pid=354 comm="systemd-backlig" name="sys"
dev="proc" ino=4026531854
scontext=system_u:system_r:systemd_backlight_t:s0-s15:c0.c1023
tcontext=system_u:object_r:sysctl_t:s0 tclass=dir permissive=0

avc:  denied  { search } for  pid=354 comm="systemd-backlig"
name="journal" dev="tmpfs" ino=1183
scontext=system_u:system_r:systemd_backlight_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

avc:  denied  { write } for  pid=354 comm="systemd-backlig" name="kmsg"
dev="devtmpfs" ino=3081
scontext=system_u:system_r:systemd_backlight_t:s0-s15:c0.c1023
tcontext=system_u:object_r:kmsg_device_t:s15:c0.c1023 tclass=chr_file
permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 2e258cd7e..65624305c 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -276,6 +276,11 @@ udev_read_runtime_files(systemd_backlight_t)
 
 files_search_var_lib(systemd_backlight_t)
 
+kernel_read_kernel_sysctls(systemd_backlight_t)
+
+mls_file_write_to_clearance(systemd_backlight_t)
+mls_file_read_to_clearance(systemd_backlight_t)
+
 #######################################
 #
 # Binfmt local policy
-- 
2.17.1

